tests/migrations/test_runner.py

```diff
@@ -209,6 +209,56 @@ def test_reverse_all() -> None:
     ), "All tables should be deleted"
 
 
+def test_reverse_idempotency_all() -> None:
+    # This test is to ensure that reversing a migration twice does not cause any
+    # issues or unintended side effects. This is important because we may need to reverse
+    # a migration multiple times in the event of a rollback.
+    runner = Runner()
+    all_migrations = runner._get_pending_migrations()
+    runner.run_all(force=True)
+    connection = get_cluster(StorageSetKey.MIGRATIONS).get_query_connection(
+        ClickhouseClientSettings.MIGRATE
+    )
+    for migration in reversed(all_migrations):
+        runner.reverse_migration(migration, force=True)
+        if migration.group != MigrationGroup.SYSTEM:
+
+            def reverse_twice() -> None:
+                # reverse again to ensure idempotency
```

Comment:
I thought the test was to run the reverse twice, e.g. `run, reverse, reverse`, instead of `run, reverse, run, reverse`?

Reply:
The run has `fake=True`. So it doesn't do anything but mark the migrations table so that reverse is not rejected immediately.